src/Form/Form.js

```diff
@@ -539,6 +542,19 @@ class Form {
 
         this.isAutofilling = false
 
+        // After autofill we check if form values match the data provided…
+        const formValues = this.getValues()
+        const areAllFormValuesKnown = Object.keys(formValues[dataType] || {}).every((subtype) => {
+            return formValues[dataType]?.[subtype] === data[subtype]
+        })
+        if (areAllFormValuesKnown) {
+            // …if we know all the values do not prompt to store data
+            this.shouldPromptToStoreData = false
+        } else {
+            // …otherwise we will prompt and do not want to autosubmit because the experience is jarring
+            this.shouldAutoSubmit = false
```

Comment:
I had it there initially but then removed it because it shouldn't always be `true`. That's bs, though, because I can just set it to `isMobileApp`, so I'll do that. 🙂